src/components/ModalInvites.tsx

```diff
@@ -0,0 +1,174 @@
+import React, { useState, useEffect, useRef } from 'react'
+import { connect, useDispatch } from 'react-redux'
+import { InviteCodes, State } from '../@types'
+import InvitesIcon from './icons/InvitesIcon'
+import CheckmarkIcon from './icons/CheckmarkIcon'
+import CopyClipboard from './icons/CopyClipboard'
+import { alert, createUserInvites } from '../action-creators'
+import { isTouch } from '../browser'
+import { theme } from '../selectors'
+import { createId, timestamp } from '../util'
+import { ActionButton } from './ActionButton'
+import Modal from './Modal'
+import Input from './Input'
+
+// eslint-disable-next-line jsdoc/require-jsdoc
+const mapStateToProps = (state: State) => {
+  const { authenticated, user: { uid = '' } = {}, userInvites } = state
+  return {
+    dark: theme(state) !== 'Light',
+    uid,
+    authenticated,
+    userInvites,
+  }
+}
+
+/** Modal to get gift codes. */
+const ModalInvites = ({ dark, uid, authenticated, userInvites }: ReturnType<typeof mapStateToProps>) => {
+  let giftCodes: { link: string; used: boolean }[] = []
+  const isMounted = useRef(false)
+  const dispatch = useDispatch()
+
+  const [inputType, updateInputType] = useState(['password', 'password', 'password'])
```

Comment:
Use an object instead of an array. It would be easier to create updates.

```javascript
const [inputType, updateInputType] = useState({
    0: 'password',
    1: 'password',
    2: 'password'
})

// You can easily create updated inpuType

const updatedInput = {
    ...inputType,
    [idToUpdate]: 'text'
} 

updateInputType(updatedInput)
```

Reply:
Changed the logic so don't need this anymore.